roxctl/declarativeconfig/create/notifier.go

```diff
@@ -0,0 +1,224 @@
+package create
+
+import (
+	"bytes"
+	"context"
+	"encoding/pem"
+	"fmt"
+	"net/url"
+	"os"
+	"sort"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/stackrox/rox/pkg/declarativeconfig"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/maputil"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/roxctl/common/environment"
+	"github.com/stackrox/rox/roxctl/declarativeconfig/k8sobject"
+	"github.com/stackrox/rox/roxctl/declarativeconfig/lint"
+	"gopkg.in/yaml.v3"
+)
+
+func notifierCommand(cliEnvironment environment.Environment) *cobra.Command {
+	notifierCommand := &notifierCmd{notifier: &declarativeconfig.Notifier{}, env: cliEnvironment}
+
+	cmd := &cobra.Command{
+		Use:   notifierCommand.notifier.Type(),
+		Args:  cobra.NoArgs,
+		Short: "Commands to create a declarative configuration for a notifier",
+	}
+
+	cmd.PersistentFlags().StringVar(&notifierCommand.notifier.Name, "name", "", "Name of the notifier")
+	cmd.AddCommand(
+		notifierCommand.genericCommand(),
+		notifierCommand.splunkCommand())
+
+	utils.Must(cmd.MarkPersistentFlagRequired("name"))
+	return cmd
+}
+
+func (n *notifierCmd) genericCommand() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "generic",
+		Args:  cobra.NoArgs,
+		RunE:  n.runE(),
+		Short: "Create a declarative configuration for a generic notifier",
+	}
+	genericFlags := cmd.Flags()
+	n.gc = &declarativeconfig.GenericConfig{}
+	genericFlags.BoolVar(&n.gc.AuditLoggingEnabled, "audit-logging", false, "Audit logging enabled")
+	genericFlags.StringVar(&n.gc.Endpoint, "webhook-endpoint", "", "Webhook endpoint URL")
+	genericFlags.StringVar(&n.gc.Username, "webhook-username", "", "Username for the endpoint basic authentication")
+	genericFlags.StringVar(&n.gc.Password, "webhook-password", "", "Password for the endpoint basic authentication")
```

Comment:
nit: could add a notion that if neither is given that no authentication will be used I suppose.